File: lib/go/internal/types/TBytes.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

const bytesName = "Bytes"

// TBytes represents the Telepact bytes type.
type TBytes struct{}

// NewTBytes constructs a TBytes instance.
func NewTBytes() *TBytes {
	return &TBytes{}
}

// GetTypeParameterCount implements TType.
func (t *TBytes) GetTypeParameterCount() int {
	return 0
}

// Validate implements TType.
func (t *TBytes) Validate(value any, typeParameters []*TTypeDeclaration, ctx *ValidateContext) []*ValidationFailure {
	return ValidateBytes(value, ctx)
}

// GenerateRandomValue implements TType.
func (t *TBytes) GenerateRandomValue(blueprintValue any, useBlueprintValue bool, typeParameters []*TTypeDeclaration, ctx *GenerateContext) any {
	return GenerateRandomBytes(blueprintValue, useBlueprintValue, ctx)
}

// GetName implements TType.
func (t *TBytes) GetName(ctx *ValidateContext) string {
	return bytesName
}

File: lib/ts/src/internal/types/TBytes.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../validation/ValidationFailure.js';
import { TTypeDeclaration } from './TTypeDeclaration.js';
import { TType } from './TType.js';
import { validateBytes } from '../validation/ValidateBytes.js';
import { generateRandomBytes } from '../generation/GenerateRandomBytes.js';
import { GenerateContext } from '../generation/GenerateContext.js';
import { ValidateContext } from '../validation/ValidateContext.js';

export const bytesName: string = 'Bytes';

export class TBytes extends TType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: TTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return validateBytes(value, ctx);
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: TTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomBytes(blueprintValue, useBlueprintValue, ctx);
    }

    getName(): string {
        return bytesName;
    }
}
